src/utilities/wait-for-task.ts

```diff
@@ -3,6 +3,12 @@ import { TaskAPI } from 'src/api';
 
 export function waitForTask(task, bailAfter = 10) {
   return TaskAPI.get(task).then((result) => {
+    const failing = ['skipped', 'failed', 'canceled'];
+
+    if (failing.includes(result.data.state)) {
+      return Promise.reject(result.data.error.description);
```

Comment:
It still doesn't need to have non-null errror...

```suggestion
      return Promise.reject(result.data.error?.description);
```